### ISSUE TITLE
useLocalStorage - globally reactivity

### ISSUE THREAD

**Issue by TangoPJ** [streamich/react-use#2607]
Hello! Is it possible to achieve reactivity of localStorage across different components when the data is updated in one of them?

Something like in this example: https://stackblitz.com/edit/react-starter-typescript-3cjkkm?file=App.tsx

**Comment by patrickhuijten** [comment]
I think `react-use` is [no longer actively being developed](https://react-hookz.github.io/web/?path=/docs/migrating-from-react-use--page).

A good alternative is `@react-hookz/web`. They cover this exact use case in [`useLocalStorageValue`](https://react-hookz.github.io/web/?path=/docs/side-effect-uselocalstoragevalue--example).

**Comment by childrentime** [comment]
I disagree with this approach. I believe that having `@react-hookz/web` handle synchronization internally is actually an anti-pattern. 

You should manage this synchronization yourself using a Context. For example:
```typescript
import { useLocalStorage } from "@reactuses/core"; // or any similar hook
import React, { createContext, useContext } from "react";

type ThemeContext = { theme: boolean; toggleTheme: () => void };

const ThemeContext = createContext<ThemeContext | undefined>(undefined);

export function ThemeProvider({ children }: { children: React.ReactNode }) {
  const [dark, setDark] = useLocalStorage('theme', false);
  
  const toggle = () => setDark(!dark);
  
  return (
    <ThemeContext.Provider value={{ theme: !!dark, toggleTheme: toggle }}>
      {children}
    </ThemeContext.Provider>
  );
}

export function useTheme() {
  const context = useContext(ThemeContext);
  if (context === undefined) {
    throw new Error("useTheme must be used within a ThemeProvider");
  }
  return context;
}
```

This gives you explicit control over the state management architecture rather than relying on hidden magic synchronization. The hook should focus on localStorage interaction, while React Context handles the global state distribution - separation of concerns.
